Add platform check to gate _fast_exit()

taranlu-houzz opened the pull request:
- Platforms that do not support using fork for multiprocessing have issues freeing
    up resources correctly when using os._exit() as opposed to sys.exit()

<!--
Before opening a PR, open a ticket describing the issue or feature the
PR will address. Follow the steps in CONTRIBUTING.rst.

Replace this comment with a description of the change. Describe how it
addresses the linked ticket.
-->

<!--
Link to relevant issues or previous PRs, one per line. Use "fixes" to
automatically close an issue.
-->

- fixes #2017 

<!--
Ensure each step in CONTRIBUTING.rst is complete by adding an "x" to
each box below.

If only docs were changed, these aren't relevant and can be removed.
-->

Checklist:

- [x] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [x] Add or update relevant docs, in the docs folder and in code.
- [x] Add an entry in `CHANGES.rst` summarizing the change and linking to the issue.
- [x] Add `.. versionchanged::` entries in any relevant code docs.
- [x] Run `pre-commit` hooks and fix any issues.
- [x] Run `pytest` and `tox`, no tests failed.

taranlu-houzz commented:
@davidism This is what I have in mind. I will run through the check list later when I have some time.

taranlu-houzz commented:
@davidism So, it seems that I need to add `platform` to the list of approved modules in order to pass `test_imports.py`. Is that alright, or is there a different way that you typically check the OS in `click`?

davidism commented:
Avoid top-level imports in Click that aren't already allowed. Instead, add the import where it's used (local instead of global).

taranlu-houzz commented:
@davidism Okay, I have run through the checklist and added a new test. I changed the platform check to be inside the `_fast_exit()` function itself. I think this is ready to be reviewed.

davidism commented:
The more I think about this, the more I think we should just drop `_fast_exit` as I suggested initially. Guarding this for "non Linux platforms" seems very heavy handed, and the fact that we're saying "skipping cleanup is safe on Linux" seems like we're just asking for trouble at some point in the future.

taranlu-houzz commented:
@davidism I created a new PR that removes `_fast_exit()` and uses `sys.exit()` instead: #2030
